ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -120,8 +123,13 @@ public void start() throws StoreException {
 
         StoreDescriptor storeDescriptor = new StoreDescriptor(dataDir);
         log = new Log(dataDir, capacityInBytes, config.storeSegmentSizeInBytes, metrics);
+        compactor =
+            new BlobStoreCompactor(dataDir, storeId, factory, config, metrics, diskIOScheduler, log, taskScheduler,
+                recovery, time, sessionId, storeDescriptor.getIncarnationId());
+        compactor.fixStateIfRequired();
         index = new PersistentIndex(dataDir, taskScheduler, log, config, factory, recovery, hardDelete, metrics, time,
```

Comment:
The index picks up index segments based on what log segments have been loaded. This can change due to the compactor fixing state (it may do a commit and cleanup). It is easier to finish such tasks and then load the index rather than load the index and then do commits and cleanups.